save-core/src/commonMain/kotlin/org/cqfn/save/core/utils/TomlUtils.kt

```diff
@@ -56,45 +51,23 @@ private inline fun <reified T : PluginConfig> Path.createPluginConfig(
  * @return list of plugin configs from toml file
  * @throws PluginException in case of unknown plugin
  */
-fun createPluginConfigListFromToml(testConfigPath: Path): MutableList<PluginConfig> {
-    val configList: MutableList<PluginConfig> = mutableListOf()
-    val parsedTomlConfig = TomlParser(testConfigPath.toString()).readAndParseFile()
-    parsedTomlConfig.getRealTomlTables().forEach { tomlPluginSection ->
-
-        // adding a fake file node to restore the structure and parse only the part of the toml
-        // this is a hack for easy partial read of Toml configuration
-        val fakeFileNode = TomlFile()
-        tomlPluginSection.children.forEach {
-            fakeFileNode.appendChild(it)
-        }
-
-        val sectionName = tomlPluginSection.name.uppercase()
-        // we don't convert sectionName to enum, because we don't want to get Kotlin exception
-        val sectionPluginConfig = when (sectionName) {
-            TestConfigSections.FIX.name -> testConfigPath.createPluginConfig<FixPluginConfig>(
-                fakeFileNode,
-                sectionName
-            )
-            TestConfigSections.FIX_AND_WARN.name -> testConfigPath.createPluginConfig<FixAndWarnPluginConfig>(
-                fakeFileNode,
-                sectionName
-            )
-            TestConfigSections.WARN.name -> testConfigPath.createPluginConfig<WarnPluginConfig>(
-                fakeFileNode,
-                sectionName
-            )
-            TestConfigSections.GENERAL.name -> testConfigPath.createPluginConfig<GeneralConfig>(
-                fakeFileNode,
-                sectionName
-            )
-            else -> throw PluginException(
```

Comment:
I think we can hold this branch in new version of `TomlUtils` too

Reply:
The exception will be thrown on the parsing of enum in case of an invalid  I do not think that we really need ‘else’ branch. That is a syntax sugar by kotlin

Reply:
Why we even just throw an exception? Where we handle this result? This will simply stop the test…

Reply:
First I thought that it is not needed, but yes, logging becomes much better with this